File: Solutions/1122. Relative Sort Array/1122. Relative Sort Array.cpp
class Solution {
public:
    vector<int> relativeSortArray(vector<int>& arr1, vector<int>& arr2) {
        vector<int> cnt(1001),res;
        for(int i=0;i<arr1.size();++i)
            ++cnt[arr1[i]];
        for(int i=0;i<arr2.size();++i){
            while(cnt[arr2[i]]--)
                res.push_back(arr2[i]);
        }
        for(int i=0;i<=1000;++i)
            while(cnt[i]-->0)
                res.push_back(i);
        return res;
    }
};

File: Solutions/1122. Relative Sort Array/1122. Relative Sort Array.java
class Solution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        int[] cnt=new int[1001];
        for(int i=0;i<arr1.length;++i)
            ++cnt[arr1[i]];
        int i=0;
        for(var n:arr2)
            while(cnt[n]-->0)
                arr1[i++]=n;
        for(int n=0;n<=1000;++n)
            while(cnt[n]-->0)
                arr1[i++]=n;
        return arr1;
    }
}